### PR TITLE
Submitting RoslynPad version 19.0.0

### PR DESCRIPTION
**Pull request by aelij**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/90739)

### CODE REVIEW

**Review by Trenly — CHANGES_REQUESTED** [review]

**Comment by Trenly** [comment]
@msftbot zip

File: manifests/r/RoslynPad/RoslynPad/17.0.0/RoslynPad.RoslynPad.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+InstallerType: zip
+Installers:
+- Architecture: x64
+  NestedInstallerType: portable
+  NestedInstallerFiles:
+  - RelativeFilePath: ./RoslynPad.exe
+    PortableCommandAlias: RoslynPad
+  InstallerUrl: https://github.com/roslynpad/roslynpad/releases/download/17/RoslynPad.zip
+  InstallerSha256: F05AB0A63E39509AF05B5636A0FB2A35E1B257A328E89F4106CECA5176C911C1
+ManifestType: singleton
```
[the remark this answers is not in the document]
Comment (aelij) [reply manifests/r/RoslynPad/RoslynPad/17.0.0/RoslynPad.RoslynPad.yaml:20]: Is that required? The docs say that's needed only for multi-locale packages:

> The singleton format is only valid for packages containing a single installer and a single locale. If more than one installer or locale is provided, the multiple YAML file format and schema must be used.

[the remark this answers is not in the document]
Comment (Trenly) [reply manifests/r/RoslynPad/RoslynPad/17.0.0/RoslynPad.RoslynPad.yaml:20]: While it still works, the multi-manifest format is preferred as it is easiest for maintaining / upgrading in the future.
https://github.com/microsoft/winget-pkgs/blob/695e87ea5e657a4b9be7446b8f8b8c8d283a2fd1/doc/manifest/schema/1.2.0/singleton.md?plain=1#L18

**Comment by aelij** [comment]
@Trenly the validation is failing because I've used schema version 1.4.0 (needed to support zip). Can this PR be merged now or do I need to wait for a non-preview release of winget that supports it?

File: manifests/r/RoslynPad/RoslynPad/17.0.0/RoslynPad.RoslynPad.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.singleton.1.4.0.schema.json
+
+PackageIdentifier: RoslynPad.RoslynPad
+PackageName: RoslynPad
+PackageVersion: 17.0.0
+PackageLocale: en-US
+PublisherUrl: https://roslynpad.net
+Publisher: Eli Arbel
+ShortDescription: C# editor and runner based on Roslyn
```
[the remark this answers is not in the document]
Comment (aelij) [reply manifests/r/RoslynPad/RoslynPad/17.0.0/RoslynPad.RoslynPad.yaml]: BTW this is not required since YAML requires a whitespace before a comment

**Comment by Trenly** [comment]
> @Trenly the validation is failing because I've used schema version 1.4.0 (needed to support zip). Can this PR be merged now or do I need to wait for a non-preview release of winget that supports it?

https://github.com/microsoft/winget-pkgs/issues/87300

@msftbot zip

**Review by ItzLevvie — DISMISSED** [review]

**Comment by Trenly** [comment]
[Policy] Reset Feedback

**Comment by Trenly** [comment]
[Policy] Dependencies

Has a runtime Dependency on Dotnet.Desktop.Runtime.7
image

**Comment by Trenly** [comment]
[Policy] Blocking-Issue

**Comment by aelij** [comment]
@Trenly What is the blocking issue? Also, why does the PR build get canceled?

**Comment by Trenly** [comment]
> @Trenly What is the blocking issue? Also, why does the PR build get canceled?

The blocking issue is that .NET Desktop Runtime is a dependency for the application to run. Even though the dependency is in the manifest, Dependency support is still experimental, so until Dependencies are in a stable 1.5 release, the dependencies are a blocking-issue

Not sure why the PR build gets canceled, it just happens sometimes and is a known issue

**Comment by aelij** [comment]
@Trenly So can I publish it without the Dependency yaml element? The app's docs indicate you have to install the .NET SDK to use it.

**Comment by Trenly** [comment]
> @Trenly So can I publish it without the Dependency yaml element?

No.

> The app's docs indicate you have to install the .NET SDK to use it.

This is exactly why it is blocked. Because you have to have the .NET SDK, the dependencies are required, but winget only supports dependencies as an experimental feature. Once dependencies are no longer experimental, this PR will be able to be merged

**Review by ItzLevvie — APPROVED** [review]

**Comment by aelij** [comment]
@ItzLevvie I've updated the package to the latest release. Can this be merged now?

**Comment by ItzLevvie** [comment]
[Policy] Dependencies

**Comment by ItzLevvie** [comment]
[Policy] Blocking-Issue

**Comment by stephengillie** [comment]
Manual Validation ended with: 
image

**Comment by stephengillie** [comment]
Hi @aelij,

This package displays the above message even with `Microsoft.DotNet.AspNetCore.8` preinstalled. Can you help me to find the correct dependency?

**Comment by aelij** [comment]
@stephengillie the app needs a .NET SDK of any supported version. It should not be a dependency, as the users can choose the version according to their needs.

**Review by stephengillie — APPROVED** [review]

**Pull request merged**